rust/processor/src/processors/token_v2_processor.rs

```diff
@@ -793,53 +796,28 @@ async fn parse_v2_token(
                             .unwrap()
                         {
                             // Add NFT ownership
-                            if let Some(inner) = TokenOwnershipV2::get_nft_v2_from_token_data(
-                                &token_data,
-                                &token_v2_metadata_helper,
-                            )
-                            .unwrap()
-                            {
-                                let (
-                                    nft_ownership,
-                                    current_nft_ownership,
-                                    from_nft_ownership,
-                                    from_current_nft_ownership,
-                                ) = inner;
-                                token_ownerships_v2.push(nft_ownership);
-                                // this is used to persist latest owner for burn event handling
+                            let (mut ownerships, current_ownerships) =
+                                TokenOwnershipV2::get_nft_v2_from_token_data(
+                                    &token_data,
+                                    &token_v2_metadata_helper,
+                                )
+                                .unwrap();
+                            if let Some(current_nft_ownership) = ownerships.first() {
```

Comment:
this took me awhile to understand. probably should rename `current_nft_ownership` to `prior_nft_ownership`, maybe add a comment saying the first value in the vec is gonna be the prior nft ownership data?

Reply:
You're not wrong. The first value is the current nft ownership. Sure I'll add a note